### PR TITLE
1.11

### PR DESCRIPTION
**Pull request by Belsnickle**

### CODE REVIEW

File: source/BatchedGeometry.h [hunk 1]
```diff
@@ -0,0 +1,195 @@
+/*-------------------------------------------------------------------------------------
```
Comment (paroj) [inline source/BatchedGeometry.h]: how did the header end up here?

**Comment by paroj** [comment]
this will require a manual merging from my side

File: source/BatchedGeometry.cpp [hunk 2]
```diff
@@ -212,16 +212,16 @@ uint32 CountUsedVertices(IndexData *id, std::map<uint32, uint32> &ibmap)
...
-   Mesh::SubMeshIterator subMeshIterator = mesh->getSubMeshIterator();
+   Mesh::SubMeshList subMeshList = mesh->getSubMeshes();
 
    // Get shared vertex data
    VertexData *oldVertexData = mesh->sharedVertexData;
 
-   while (subMeshIterator.hasMoreElements()) {
-      SubMesh *subMesh = subMeshIterator.getNext();
+   for (auto subMeshIter = subMeshList.begin(); subMeshIter != subMeshList.end(); ++subMeshIter) {
+	   SubMesh *subMesh = subMeshList.front();
```
Comment (paroj) [inline source/BatchedGeometry.cpp]: this is wrong

File: source/ImpostorPage.cpp [hunk 3]
```diff
@@ -445,20 +445,20 @@ void ImpostorTexture::updateMaterials()
...
-	texture.setNull();
+	texture.reset();
 	if (TextureManager::getSingletonPtr())
 		TextureManager::getSingleton().remove(texName);
 	
 	//Delete materials
 	for (int o = 0; o < IMPOSTOR_YAW_ANGLES; ++o){
 	for (int i = 0; i < IMPOSTOR_PITCH_ANGLES; ++i){
-		assert (!material[i][o].isNull());
+		assert (!material[i][o]);
```
Comment (paroj) [inline source/ImpostorPage.cpp]: this is wrong

**Comment by paroj** [comment]
contains too many errors to be merged - even manually. The chances are high that I miss something.

File: source/BatchedGeometry.h (hunk 1 above)
Comment (Belsnickle) [reply source/BatchedGeometry.h]: First time using a source control system like GitHub.  I copied all files that I edited into the source directory of my pull request.  I removed it from the branch and then uploaded it to the include folder.

**Comment by paroj** [comment]
feel free to create a reduced version of this with a minimal amount of tested changes.
- some static_pointer_casts are not needed. e.g. `Ogre::MaterialManager::getSingleton().getByName` already returns a `MaterialPtr` since 1.9
- watch out for `!foo.isNull` becoming `!foo` which is wrong and hard to spot

**Pull request closed without merging**